### PR TITLE
Fix up the C++ snippet to reflect the carbon snippet

### PR DESCRIPTION
**Pull request by PJBoy**
Offers a more precise apples-to-apples language comparison. Replaces the non-portable `M_PI` and removes the extraneous arg parameters. It does use C++23, so I can understand if that's unacceptable for the time being.

### CODE REVIEW

File: docs/images/snippets.md
Comment (zygoloid) [inline docs/images/snippets.md]: I think the `const&` here is a slightly better match for the Carbon code, though neither is exactly right -- the Carbon code doesn't force a copy to be made, but it has value semantics rather than reference semantics.

**Comment by jonmeow** [comment]
Note, the CLA is required. But glancing at the change, I don't feel it's an unambiguous choice; I'll let one of the leads decide whether they want to adopt these changes (note, the images would also need to be updated).

File: docs/images/snippets.md
Comment (zygoloid) [inline docs/images/snippets.md:130]: It'd be good to consistently use `import` here too, at least in place of `#include <vector>`.

If you wanted to update the other includes, we don't have rules yet for how to map Carbon libraries to C++ module names, but `import Carbon:Geometry;` seems like a good guess to use as a placeholder. Or you could use `import "geometry.carbon.h";` for now.

File: docs/images/snippets.md
Comment (PJBoy) [reply docs/images/snippets.md]: Thanks for the feedback. Agree that `const&` is a more faithful approximation

File: docs/images/snippets.md
Comment (PJBoy) [reply docs/images/snippets.md:130]: I have opted for the latter choice. I felt the former choice would be more appropriate if we wrote circle.h to export a module (say `export module circle;`), but then I didn't want to guess what the consuming Carbon code would look like (`import Cpp module circle;`?)

**Review by zygoloid — COMMENTED** [review]
Sorry for the latency here. @chandlerc and I have talked about this a bit, and we'd like to hold off on using `import` in the C++ examples for now. While we eventually will want to write this example this way, we think the majority of C++ developers aren't yet able to use `import std;`, so using that in the examples is probably going to be a distraction more than it's helpful.

What do you think about changing the examples to something more like:

```c++
#include <numbers>
#include <print>
#include <span>
#include <stdfloat>
#include <vector>
// or: import std;
```

for the time being?

**Review by zygoloid — APPROVED** [review]
Thank you!

**Pull request merged**
